src/cli/type-generator.js

```diff
@@ -108,10 +114,130 @@ module.exports = class TypeGenerator {
+      subgraph.get('dataSources').map(dataSource => {
+        dataSource.getIn(['mapping', 'abis']).map(abi => {
+          allFiles.push(abi.get('file'))
+        })
+      })
+
+      // Make paths absolute
+      let allAbsolutePaths = allFiles.map(file => {
+        return path.resolve(this.sourceDir, file)
+      })
+      return allAbsolutePaths
+    } catch (e) {
+      this.logger.fatal('Failed to parse subgraph file locations:', e)
+    }
+  }
+
+  watchAndGenerateTypes() {
+    let generator = this
+    if (cluster.isMaster) {
+      let worker = cluster.fork()
+      worker.send({msgFromMaster: 1});
+      cluster.on('exit', function () {
+        cluster.fork()
+      })
+    }
+
+    if (cluster.isWorker) {
+      // Use master message to kick off compile
+      // preventing infinite error loop
+      process.on('message', function (_) {
+        generator.generateTypes()
+      });
+      let watchedFiles = this.getFilesToWatch()
+
+      // Initialize watcher that reruns generator
+      let watcher = chokidar.watch(this.options.subgraphManifest, {
+        persistent: true,
+        ignoreInitial: true,
+        atomic: 500
+      })
+      watcher
+        .on('ready', function () {
+          generator.logger.info('')
+          generator.logger.info(chalk.grey('Watching relevant manifest files'))
+        })
+        .on('change', changedFile => {
+          generator.logger.info('')
+          generator.logger.info('%s %s',
+            chalk.grey('File change detected: '),
+            generator.displayPath(changedFile)
+          )
+
+          // Convert to absolute paths for comparison
+          let manifestAbsolute =
+            (path.resolve(generator.options.subgraphManifest) === path.normalize(generator.options.subgraphManifest))
+            ? this.options.subgraphManifest
+            : path.resolve(generator.sourceDir, generator.options.subgraphManifest)
+          let changedFileAbsolute = (path.resolve(changedFile) === path.normalize(changedFile))
+            ? changedFile
+            : path.resolve(generator.sourceDir, changedFile)
+
+          if (changedFileAbsolute === manifestAbsolute) {
+            // Update watcher based on changes to manifest
+            let updatedWatchFiles = generator.getFilesToWatch()
+            let addedFiles = updatedWatchFiles.filter(file => {
+              return watchedFiles.indexOf(file) === -1
+            })
+            let removedFiles = watchedFiles.filter(file => {
+              return updatedWatchFiles.indexOf(file) === -1
+            })
+            watchedFiles = updatedWatchFiles
+            watcher.add(addedFiles)
+            watcher.unwatch(removedFiles)
+
+            if (addedFiles.length >= 1) {
+              let addedFilesDisplay = addedFiles.map(file => {
+                return generator.displayPath(file)
+              })
+              generator.logger.info(
+                '%s %j',
+                chalk.grey('Now watching: '),
```

Comment:
Several remarks here:

1. Let's use the debugging level for these kinds of messages.
2. Let's format the files nicely, the current output is not user-friendly. How about the following format?
```
Now watching:
  - abis/Foo.json
  - mappings/bar.ts
```

Reply:
sounds good :)